default.nix

```diff
@@ -71,40 +71,8 @@ let
     (import "${hnix-store-src}/overlay.nix")
     (self: super: with pkgs.haskell.lib; {
 
-      # Type error in the tests under ghc844 package set
-      Diff = dontCheck super.Diff;
-
-      # These packages only depend on contravariant when ghc >= 8.6.3
-      # Without normalizing the dependencies, our build fails with
-      # aeson and base-compat-batteries unable to find `contravariant`
-      aeson                 = addBuildDepend super.aeson self.contravariant;
-      base-compat-batteries = addBuildDepend super.base-compat-batteries self.contravariant;
-
-      mono-traversable  = dontCheck super.mono-traversable;
-      regex-tdfa-text   = doJailbreak super.regex-tdfa-text;
-      these             = doJailbreak super.these;
       semialign         = super.semialign_1_1;
```

Comment:
Do you remember why this one is needed? I think I didn't need it for `GHC-8.8.1` nor `8.10.1` but haven't tested on `8.6.5`. Just for the reference.

Reply:
IDK/remember what was asking, not paid attention into it, just remember `8.6.5` was asking for it, so I left it.
First cleaning-up from all easy no longer needed stuff, so we would know what is left is indeed needed for some reason.